src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -345,6 +349,30 @@ String getTopicNameFromNamespace(final String prefix, final BsonDocument namespa
   }
 
   Map<String, Object> createPartitionMap(final MongoSourceConfig sourceConfig) {
+    return createPartitionMap(sourceConfig, partitionMap == null);
+  }
+
+  private Map<String, Object> createPartitionMap(final MongoSourceConfig sourceConfig, final boolean recreate) {
+    if (recreate) {
+      ConnectionString connectionString = sourceConfig.getConnectionString();
+      StringBuilder builder = new StringBuilder();
```

Comment:
What would you think of actually putting this logic in the Java driver itself in the future? It seems like something that could be used in multiple places at some point.

Reply:
I could see a ConnectionString#getClean() method that did something similar, but the database/collection config seems specific to this use case.

Reply:
Ideally the toString method really shouldn't leak the sensitive information, but users may rely on it providing the original string.